okhttp/src/main/java/okhttp3/internal/connection/RouteSelector.kt

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (C) 2012 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal.connection
+
+import okhttp3.Address
+import okhttp3.Call
+import okhttp3.EventListener
+import okhttp3.HttpUrl
+import okhttp3.Route
+import okhttp3.internal.Util
+import java.io.IOException
+import java.net.InetSocketAddress
+import java.net.Proxy
+import java.net.SocketException
+import java.net.UnknownHostException
+import java.util.NoSuchElementException
+
+/**
+ * Selects routes to connect to an origin server. Each connection requires a choice of proxy server,
+ * IP address, and TLS mode. Connections may also be recycled.
+ */
+class RouteSelector(
+  private val address: Address,
+  private val routeDatabase: RouteDatabase,
+  private val call: Call,
+  private val eventListener: EventListener
+) {
+  /* State for negotiating the next proxy to use. */
+  private var proxies = emptyList<Proxy>()
+  private var nextProxyIndex: Int = 0
+
+  /* State for negotiating the next socket address to use. */
+  private var inetSocketAddresses: List<InetSocketAddress> = emptyList()
```

Comment:
```suggestion
  private var inetSocketAddresses = emptyList<InetSocketAddress>()
```
You're mixing with type / without type